pkgs/build-support/dart/build-dart-package/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib, stdenv, fetchDartDeps, writeText, dartHooks, dart, nodejs }:
+
+{ pubGetScript ? "dart pub get"
+
+  # Output type to produce. Can be any kind supported by dart
+  # https://dart.dev/tools/dart-compile#types-of-output
+  # If using jit, you might want to pass some arguments to `dartJitFlags`
+, dartOutputType ? "exe"
```

Comment:
I have very limited knowledge about dart (all of it comes from packaging dart-sass-embedded), but now that this is supporting AOT builds, should it be the default?

The standalone exe build will embed a copy of dart, which seems a bit redundant in the context of a package manager like Nix.

Reply:
Oh maybe exe would make as a default for `stdenv.hostPlatform.isStatic` though!

Reply:
Thanks a lot for the feedback!

I think it makes sense doing `aot` by default, yeah:
```sh
# static
$ du -h /nix/store/vc2b4w8vlkbvahcipa2d8sp7byv9hwhd-dart-sass-1.62.1/bin/dart-sass
5.0M	/nix/store/vc2b4w8vlkbvahcipa2d8sp7byv9hwhd-dart-sass-1.62.1/bin/dart-sass
# aot
$ du -h /nix/store/qyk4nx1zn3aaaf22qay9xhcf7aan8c8h-dart-2.19.6/bin/dartaotruntime /nix/store/8823mp1vjdz58804xl860aczq5wkfh6w-dart-sass-1.62.1/share/bin/dart-sass
5.0M	/nix/store/qyk4nx1zn3aaaf22qay9xhcf7aan8c8h-dart-2.19.6/bin/dartaotruntime
4.1M	/nix/store/8823mp1vjdz58804xl860aczq5wkfh6w-dart-sass-1.62.1/share/bin/dart-sass
```

I like that `isStatic` suggestion a lot, really sweet stuff. I'll get it done in a sec

Reply:
Actually, hmm, I think it's better to use static builds until we get a separate `dart-runtime` package. The dart derivation is reaaally big:
```sh
$ du -sh /nix/store/qyk4nx1zn3aaaf22qay9xhcf7aan8c8h-dart-2.19.6/bin/
551M	/nix/store/qyk4nx1zn3aaaf22qay9xhcf7aan8c8h-dart-2.19.6/bin/
```